test/foundry/deploy/15_holograph_registry_tests.t.sol

```diff
@@ -0,0 +1,802 @@
+        // identically to testAllowAdminToAlter_holographSlot().
+        vm.skip(true);
+        assertEq(holographRegistry.getHolograph(), mockAddress);
+    }
+
+/**
+ * @notice Test allowing an external contract to call the getHolograph function.
+ * @dev This test verifies that an external contract can successfully call the getHolograph function
+ * of the registry contract without any parameters.
+ * It simulates an external contract calling the getHolograph function in the registry.
+ * Refers to the hardhat test with the description 'Should allow external contract to call fn'
+ */
+    function testAllowExternalContractToCallFnGetHolograph() public {
+        bytes memory encodeSignature = abi.encodeWithSignature('getHolograph()');
+        mockExternalCall.callExternalFn(address(registry), encodeSignature);           
+    }
+
+/*
+    setHToken()
+*/
+
+/**
+ * @notice Test allowing an admin to alter the hTokens mapping in the registry.
+ * @dev This test verifies that an admin can successfully update the hToken address for a specific chain ID
+ * in the registry with a new address.
+ * It simulates the admin calling the setHToken function with a valid chain ID and hToken address,
+ * and then checks if the registry correctly stores and returns the updated hToken address for the chain ID.
+ */
+    function testAllowAdminToAlter_hTokens() public {
+        vm.startPrank(holographRegistry.getAdmin());
+        holographRegistry.setHToken(validChainId, hTokenAddress);
+        assertEq(holographRegistry.getHToken(validChainId), hTokenAddress);
+        vm.stopPrank();
+    }
+
+/**
+ * @notice Test the revert when the owner tries to alter the hTokens mapping in the registry.
+ * @dev This test verifies that the registry reverts with the message 'HOLOGRAPH: admin only function'
+ * when the owner (not the admin) attempts to alter the hToken address for a specific chain ID.
+ * It simulates the owner trying to call the setHToken function with a valid chain ID and hToken address.
+ * Refers to the hardhat test with the description 'should fail to allow owner to alter _hTokens'
+ */
+    function testAllowOwnerToAlter_hTokensRevert() public {
+        vm.expectRevert('HOLOGRAPH: admin only function');
+        holographRegistry.setHToken(validChainId, hTokenAddress);
+    }
+
+/**
+ * @notice Test the revert when a non-admin user tries to alter the hTokens mapping in the registry.
+ * @dev This test verifies that the registry reverts with the message 'HOLOGRAPH: admin only function'
+ * when a non-admin user attempts to alter the hToken address for a specific chain ID.
+ * It simulates a random user calling the setHToken function with a valid chain ID and hToken address.
+ * Refers to the hardhat test with the description 'should fail to allow non-owner to alter _hTokens'
+ */
+    function testAllowNonOwnerToAlter_hTokensReturn() public {
+        vm.prank(randomAddress());
```

Comment:
same like in line [634](https://github.com/holographxyz/holograph-protocol/pull/314/files#diff-cc277e07cad54b5577787ca9c4f8cbae41c4229c76c3792864829d3f93a6576cR634)